### ISSUE TITLE
React Native support for iOS and Android

### ISSUE THREAD

**Issue by mehrdadfeller** [jaxcore/bumblebee-hotword#1]
Sorry if this is a stupid question: I was wondering I can use this directly in React Native App on iOS or Android easily? Do I need to do anything to get the microphone/audio working (permission-wise) on these platforms?

**Comment by mehrdadfeller** [comment]
Perpahs I should use something like this:

https://github.com/chadsmith/react-native-microphone-stream

**Comment by jaxcore** [comment]
Yes I think it would work, but you'd have to figure out how to take that React audio stream event:

```
const listener = MicStream.addListener(data => console.log(data));
```

And wire that event to this library by replacing/rewriting the picovoiceAudioManager which is essentially just a Web Audio API stream.

https://github.com/jaxcore/bumblebee-hotword/blob/master/lib/picovoiceAudioManager.js

It would take a bit of fiddling with those process callbacks, and it's a little confusing even to me.

The first thing I'd try is something like this starting at line 72:

```
        //navigator.getUserMedia({audio: true}, getUserMediaSuccessCallback, errorCallback);

        engine = picovoiceEngine;
        
        //processCallback = picovoiceProcessCallback;

      MicStream.addListener((data) => {
         engine.process(data);
      })
```

If you can figure it out I'd be happy to accept a PR.

**Comment by jaxcore** [comment]
Oh, you're gonna run into some WASM loading problems too.  See this:

https://github.com/ExodusMovement/react-native-wasm
